### PR TITLE
DON-856 Add configuration for drop down width.

### PR DESCRIPTION
**Pull request by henrik-sky**
[Update](https://skyscanner.slack.com/archives/C07LH0LRCAK/p1730104608155989) corner radius for BpkSelect to match [designs](https://www.figma.com/design/irZ3YBx8vOm16ICkAr7mB3/Backpack-Components?node-id=6358-6521&t=9Dk8GpPy5v7oXhNc-4).

New parameter! 

```
enum class BpkDropDownWidth {
MaxWidth, MatchOptionWidth, MatchSelectWidth
}
```

| MaxWidth | MatchOptionWidth | MatchSelectWidth |
| ------------ |------------ |------------ |
| Screenshot_20241031_113458 | Screenshot_20241031_113509 | Screenshot_20241031_113519  |

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/textfield/internal/BpkTextFieldImpl.kt [hunk 1]
```diff
@@ -333,12 +333,13 @@ private fun RowScope.TrailingIcon(
 internal enum class BpkTextFieldType {
     Default,
     Search,
+    Select,
```
Comment (bvitaliyg) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/textfield/internal/BpkTextFieldImpl.kt:336]: So if that's the case, anyone can create a TextField with Select appearance even though it's only supposed to be used internally. If it's not a part of the API design we have in Figma, it should be done via internal mappings instead.

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/select/BpkSelect.kt [hunk 2]
```diff
@@ -32,8 +33,9 @@ fun BpkSelect(
     modifier: Modifier = Modifier,
     status: BpkFieldStatus = LocalFieldStatus.current,
     onSelectionChange: ((selectedIndex: Int) -> Unit)? = null,
+    dropDownWidth: BpkDropDownWidth = BpkDropDownWidth.MaxWidth,
```
Comment (bvitaliyg) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/select/BpkSelect.kt:36]: Is this a part of the Figma API upgrade?

**Review by bvitaliyg — COMMENTED** [review]
Thank you for the contribution! Have a few questions

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/textfield/internal/BpkTextFieldImpl.kt (hunk 1 above)
Comment (henrik-sky) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/textfield/internal/BpkTextFieldImpl.kt:336]: `BpkTextFieldType` is only used in `BpkTextFieldImpl` so not available to consumers was my understanding

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/select/internal/BpkSelectImpl.kt [hunk 3]
```diff
@@ -39,11 +40,17 @@ import net.skyscanner.backpack.compose.fieldset.LocalFieldStatus
...
 import net.skyscanner.backpack.compose.text.BpkText
 import net.skyscanner.backpack.compose.textfield.internal.BpkTextFieldImpl
+import net.skyscanner.backpack.compose.textfield.internal.BpkTextFieldType
 import net.skyscanner.backpack.compose.theme.BpkTheme
 import net.skyscanner.backpack.compose.tokens.ArrowDown
 import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.utils.applyIf
 import net.skyscanner.backpack.compose.utils.clickableWithRipple
 
+enum class BpkDropDownWidth {
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/select/internal/BpkSelectImpl.kt:50]: this should not be defined in the internal package if it is exposed publicly

Comment (henrik-sky) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/select/internal/BpkSelectImpl.kt]: Ah yeah good spot

**Review by marianeum — APPROVED** [review]

**Pull request merged**